Services/User/classes/class.ilObjUser.php

```diff
@@ -212,6 +212,7 @@ public function __construct($a_user_id = 0, $a_call_by_reference = false)
             $this->setId($a_user_id);
             $this->read();
         } else {
+            $this->setId(ANONYMOUS_USER_ID);
```

Comment:
Please note that @pascalseeland takes care of the authentication service, but not of the user server, which is still maintained by @smeyer-ilias.

This should not be fixed in the user class. The ID is usually set to 0 when new objects (here new users) are created. It is wrong to set the anonymous ID in these processes automatically. New user objects should not get the anonymous user ID. Whatever code calls the constructor here during initialisation should already pass the anonymous user ID.